src/include/mpir_func.h

```diff
@@ -21,51 +21,47 @@
 
 /* function enter and exit macros */
 #if defined(MPL_USE_DBG_LOGGING)
-#define MPIR_FUNC_ENTER(a) MPL_DBG_MSG(MPL_DBG_ROUTINE_ENTER,TYPICAL,"Entering "#a)
+#define MPIR_FUNC_ENTER MPL_DBG_MSG_S(MPL_DBG_ROUTINE_ENTER,TYPICAL,"Entering %s", __func__)
 #elif defined(MPICH_DEBUG_MEMARENA)
-#define MPIR_FUNC_ENTER(a) MPL_trvalid("Entering " #a)
+#define MPIR_FUNC_ENTER MPL_trvalid2("line %d - entering %s\n", __LINE__, __func__)
```

Comment:
I notice we are adding line number here in the log output. Not opposed to that change, just curious if this is purely additive, or is it replacing some previous behavior that is not obvious?

Reply:
We need pass `__func__` as a non-literal string, which forces us to use `MPL_trvalid2`, which is not a macro and takes fixed parameters that include a line number.